### ISSUE TITLE
navbar-right and navbar-left are only functional inside of a navbar-collapse

### ISSUE THREAD

**Issue by erebel55** [twbs/bootstrap#18411]
The navbar-right and navbar-left css classes only work correctly if inside of a navbar-collapse. This isn't noted in the docs. However, we can easily see that it is true by looking at navbar.less lines 378 - 388.

```css
@media (min-width: @grid-float-breakpoint) {
  .navbar-left  { .pull-left(); }
  .navbar-right {
    .pull-right();
    margin-right: -@navbar-padding-horizontal;

    ~ .navbar-right {
      margin-right: 0;
    }
  }
}
```

This only pulls the navbar-right/left classes if the breakpoint is it. However, if we aren't using navbar-collapse we don't care about the breakpoint.

An example jsbin can be found here:
http://jsbin.com/rirese/edit?html,css,output

If you shrink the view to less than the breakpoint (768px) you will see the havoc.

We could possibly use :not css selector to find navbar-left/right's that are NOT a descendant of navbar-collapse and perform this pull-right/pull-left regardless of the breakpoint.

**Comment by twbs-lmvtfy** [comment]
Hi @erebel55!

You appear to have posted a live example (http://jsbin.com/rirese/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 12, column 9 thru column 76: Bad value ```` navigate ```` for attribute ```` role ```` on element ```` nav ````.
* line 18, column 33 thru column 49: Element ```` img ```` is missing required attribute ```` src ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by erebel55** [comment]
Here is the revised jsbin example
http://jsbin.com/divuki/edit?html,css,output

**Comment by twbs-lmvtfy** [comment]
Hi @erebel55!

You appear to have posted a live example (http://jsbin.com/divuki/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness
* line 6, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.5
* line 8, column 5: [W013](https://github.com/twbs/bootlint/wiki/W013): Bootstrap version might be outdated. Latest version is at least 3.3.6 ; saw what appears to be usage of Bootstrap 3.3.5

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by erebel55** [comment]
Sorry about that, here is an updated jsbin.
http://jsbin.com/jozodun/edit?html,css,output

**Comment by cvrebert** [comment]
Closing as WontFix per https://github.com/twbs/bootstrap/issues/18362#issuecomment-160498098
See also https://github.com/twbs/bootstrap/issues/17598#issuecomment-160248326

**Comment by erebel55** [comment]
Thank you!
